src/test/ui/issues/issue-44056.rs

```diff
@@ -10,6 +10,7 @@
 
 // compile-pass
 // only-x86_64
+// gate-test-avx512_target_feature
```

Comment:
`gate-test` is AFAIK used to indicate that this is a "feature gate test", to check the policy that every feature gate comes with a test making sure stuff is indeed gated.

You want some kind of `only-` annotation here to run the test only when certain conditions are met.

Reply:
`only` and `ignore` accept the same flags, maybe you can grep for `ignore`?

The code for this is at https://github.com/rust-lang/rust/blob/0117b42f66632f1f0fc08f1cdd8ca264c10bba94/src/tools/compiletest/src/header.rs#L726. Maybe you need to add a way to control test execution based on CPU flags.